pkg/controller/servicebindingrequest/olm.go

```diff
@@ -142,13 +156,113 @@ func (o *OLM) SelectCRDByGVK(gvk schema.GroupVersionKind) (*olmv1alpha1.CRDDescr
 		return nil, err
 	}
 
-	if len(crdDescriptions) == 0 {
+	if len(crdDescriptions) == 0 && crdDescription == nil {
 		log.Debug("No CRD could be found for GVK.")
 		return nil, fmt.Errorf("no crd could be found for gvk")
+	} else if len(crdDescriptions) == 0 {
+		// use the crdDescription built from CRD annotations as fallback
+		return crdDescription, nil
 	}
+
 	return crdDescriptions[0], nil
 }
 
+// buildCRDDescriptionFromCRDAnnotations builds a CRDDescription from annotations present in the CRD.
+func buildCRDDescriptionFromCRDAnnotations(crd *unstructured.Unstructured) (*olmv1alpha1.CRDDescription, error) {
+	var specDescriptors []olmv1alpha1.SpecDescriptor
+	var statusDescriptors []olmv1alpha1.StatusDescriptor
+
+	specDescriptors, statusDescriptors, err := buildDescriptorsFromAnnotations(crd.GetAnnotations())
+	if err != nil {
+		return nil, err
+	}
+
+	kind, ok, err := unstructured.NestedString(crd.Object, "spec", "names", "kind")
+	if err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil
+	}
+
+	version, ok, err := unstructured.NestedString(crd.Object, "spec", "version")
+	if err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil
+	}
+
+	crdDescription := olmv1alpha1.CRDDescription{
+		Name:              crd.GetName(),
+		Version:           version,
+		Kind:              kind,
+		SpecDescriptors:   specDescriptors,
+		StatusDescriptors: statusDescriptors,
+	}
+
+	return &crdDescription, nil
+}
+
+func buildDescriptorsFromAnnotations(annotations map[string]string) ([]olmv1alpha1.SpecDescriptor, []olmv1alpha1.StatusDescriptor, error) {
```

Comment:
I'm missing the documentation comment on this method, and apparently, the amount of columns in this line is higher than 101 (defined by `.editorconfig`).

Reply:
Solved in da312bdda4fcf1525cd9a40168d50247a4dde05e.